### PR TITLE
A test for Tgz#name

### PR DESCRIPTION
**Pull request by Sammers21**
Related to #19

Can be merged only after #23

### CODE REVIEW

**Comment by Sammers21** [comment]
@g4s8, pay attention to the Pull Request.

**Comment by 0crat** [comment]
This pull request [#24](https://github.com/artipie/helm-adapter/issues/24) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/6988add1-22ca-4ec2-91eb-3d405a3ed42b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/helm/TgzArchive.java [hunk 1]
```diff
@@ -61,10 +68,22 @@
...
-        //  For now this method is not implemented. We should decompress the archive, find the chart
-        //  version and name, and, return it, binded together.
-        throw new IllegalStateException("Not Implemented");
+        try {
+            final TarArchiveInputStream taris = new TarArchiveInputStream(
+                new GzipCompressorInputStream(new ByteArrayInputStream(this.content))
+            );
+            TarArchiveEntry entry;
+            while ((entry = taris.getNextTarEntry()) != null) {
+                if (!entry.getName().endsWith("Chart.yaml")) {
```
Comment (paulodamaso) [inline src/main/java/com/artipie/helm/TgzArchive.java:77]: @Sammers21 I think that the following would be more readable:
```
while(...) {
    if (entry.getName().endsWith("Chart.yml") {
                final Map<String, Object> load = new Yaml().load(taris);
                return String.format("%s-%s.tgz", load.get("name"), load.get("version"));
    }
}
```

Also, the code would execute as intended if we had a file named "PirateChart.yml" inside the Tar archive?

Comment (Sammers21) [reply src/main/java/com/artipie/helm/TgzArchive.java:77]: @paulodamaso, it is a part of #23

Comment (Sammers21) [reply src/main/java/com/artipie/helm/TgzArchive.java:77]: @paulodamaso, but thank you for the comment, I've fixed it

**Comment by paulodamaso** [comment]
@g4s8 please merge

**Comment by Sammers21** [comment]
@paulodamaso, I will restrict the merge until #23 merge

**Comment by Sammers21** [comment]
@g4s8, we can merge it now

**Comment by 0crat** [comment]
Job was finished in 20 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@paulodamaso please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@sereshqua I will, thank you

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**